### ISSUE TITLE
Catalog update not indexing according to file ID3 tags

### ISSUE THREAD

**Issue by arno1979** [ampache/ampache#3289]
## Description
I just added the album "Men (Original Motion Picture Soundtrack)" to my catalog and ran the catalog update from CLI to add the album to the database. But artist, album artist and song artist are not added to the database according the files' tags.

### Describe the bug
According to the ID3 tags of the album's mp3 files the tags "artist" and "album artist" should both be "Ben Salisbury & Geoff Barrow" (as per the musicbrainz.org entry for the album: https://musicbrainz.org/release/7d2daf54-2483-4677-a97a-602a963334f5).
But when I check the new entry via the web interface (or my Subsonic client on my phone) the album's artist is given as "Ben Salisbury" only. Geoff Barrow is omitted. In the album view of the web interface I can see that for each song the song artist has two entries: "Unknown (Orphaned)" and "Ben Salisbury".
As I have other albums from these two artists working together (all tagged as "Ben Salisbury & Geoff Barrow") and they are being indexed correctly I have no clue why it is not working for this specific album.

I have to add that I use musicbrainz.org as the source for creating the tags (via the Yate app on my Mac). And the information in the musicbrainz database for Ben Salisbury and Geoff Barrow working together is not very unified. They are credited as "Ben Salisbury & Geoff Barrow", "Ben Salisbury / Geoff Barrow", "Ben Salisbury and Geoff Barrow" among other variations.

So I tried to unify the tags manually (again using Yate) so that all their albums are indexed under "Ben Salisbury & Geoff Barrow. This works in most cases but not for the "Men" soundtrack (and also not for the "Archive 81" soundtrack, but that album is not listed on musicbrainz.org at all). Both these albums are indexed in the way described above (album artist "Ben Salisbury", song artist "orphaned" + "Ben Salisbury").

I noticed another strange indexing behavior:
When adding the soundtrack album for Ex_Machina, after the first catalog update the album artist is displayed as "Ben Salisbury / Geoff Barrow" (which is wrong and nowhere in the tags of the album's files is the artist ever given as "Ben Salisbury / Geoff Barrow"). But when indexing the catalog for the second time after adding the album to the catalog's directory the album artists is corrected and given as "Ben Salisbury & Geoff Barrow".

One more info: I tested all this by moving the album's files out of the catalog directory each time and updating the catalog before moving the files back into the catalog directory. Therefore making sure that the info hopefully gets correctly removed from the database.

### Expected behavior
The album should be indexed according to the ID3 tags

## Environment

* Ampache version: 5.4.0
* Web server + version: Apache 2.4.53
* Server operating system: Linux Fedora 36
* Client operating system: macOS 12.4

### Client type

* Web
* SubSonic

## Settings

From my `ampache.cfg.php`:
`metadata_order = "getID3,filename"`

**Comment by lachlan-00** [comment]
are they mp4/aac audio? they is an issue with the tag reader we use and arrays in aac tags

**Comment by arno1979** [comment]
Yes, I have one album in the bunch that's encoded as AAC (in m4a containers). Funnily enough that album is not causing any problems. At least as far as I can tell.

**Comment by lachlan-00** [comment]
ok, there are 2 debug areas that will be important
`src/Repository/Model/Catalog.php` the debug line here prints all the results from reading your tags
![image](https://user-images.githubusercontent.com/1305249/172318355-98c623cd-c444-4338-af6e-1ccfe326842e.png)

`src/Module/Util/VaInfo.php` has debug for each type and is very loud so i wouldn't look at that until after you print your results

each type of tag is commented out  using the newer logger methods
```
//$this->logger->debug
```

if debugging and logs are too much you can put an  example file up somewhere and send me a link privately

**Comment by arno1979** [comment]
Okay, thanks for looking into it!
I'll have to look at the files when I'm home and have a little time.
I will get back to you as soon as I have some log files for you to look at.

**Comment by arno1979** [comment]
Hi lachlan,

I did as you suggested and activated the debug line in Catalog.php.
I attach two files: the first is the inital catalog update after I first removed and then re-added the files in question.
The second file is a second catalog update. I thought maybe this would enable you to figure out why a change of artist happens in the case of the Ex_Machina album.

[catalog-update-first-run.log](https://github.com/ampache/ampache/files/8855681/catalog-update-first-run.log)
[catalog-update-second-run.log](https://github.com/ampache/ampache/files/8855682/catalog-update-second-run.log)

**Comment by lachlan-00** [comment]
that's great i'll be able to read through this in a few hours.

it will be good to find out if it's anything weird / logic based happening as we're not seeing a lot of these

**Comment by lachlan-00** [comment]
just starting now, are you running php8.1? i see a lot of random errors for depreciations that might be the cause.

It looks to me like the tags are being read correctly at all times but then the processing is either emptying/not setting the data correctly.

I think it might be time for me to update to php8.1

**Comment by arno1979** [comment]
> just starting now, are you running php8.1? i see a lot of random errors for depreciations that might be the cause.
Yes, I am. I recently updated my server to Fedora 36 and PHP 8.1 is the default version for that release.

> It looks to me like the tags are being read correctly at all times but then the processing is either emptying/not setting the data correctly.

I think so too. What I did see when glancing at the log files was that tags were being read that shouldn't be there. Best example would be the artist string "Ben Salisbury / Geoff Barrow". As far as I know that tag is nowhere present in the files. At least not in any field that is visible through my tag editor.
I did notice though (I think it was the "Men" album) that the artist string was read as "Ben Salisbury & Geoff Barrow". But just under that info an artist array was displayed. This array only contained one entry: "Ben Salisbury". I thought that maybe it was a case where the indexer was trying to split the string "Ben Salisbury & Geoff Barrow" into multiple artists but was having trouble with the "&"? The same thing could happen with "Ben Salisbury / Geoff Barrow" because of the "/".

One more thought: I have mbid entries embedded in nearly all these files ("Archive 81" is not in the musicbrainz database, therefore no entries there). Is it possible that the indexer is using the mbid to read artist info from the musicbrainz database? As far as I know the "DROKK" album is credited to "Ben Salisbury / Geoff Barrow" on musicbrainz.
I was trying to unify the artist to "Ben Salisbury & Geoff Barrow" for these releases in my ampache database so I manually changed the artists on some albums. Just a tought...

**Comment by lachlan-00** [comment]
yes the mbid's for artists are generated into the first item and an array.

* [mb_albumartistid] & [mb_albumartistid_array] 
* [mb_artistid] & [mb_artistid_array]

If these are empty it's done based on the name ([albumartist] / [artist])

If they are mbid's that don't match an existing artist we do a lookup to musicbrainz for the artist name
eg. lookup 2ef8e9b2-5f7d-4cfb-899a-bc4544755371 and get the artist Sadness.

If your tags i don't see any artist mbids so it should all go off names 

BUT i do see a lot of empty artists in the artists array (this was from /mnt/music/TEST/Men OST - [2022]/12. Birth.mp3)
![image](https://user-images.githubusercontent.com/1305249/172810106-ab240b60-88cc-4246-825c-8353c85b1186.png)

**Comment by lachlan-00** [comment]
I think there are some logic workarounds that need to happen

your artists array is 4 items which overrides the other sections and probably creates those empty items

**Comment by arno1979** [comment]
I will close this issue as I found out that the cause for the problem was due to some unclean tagging done on my side.
I figured out a way to view all tags of a selected file and there I found that some of my files hat two "artist" tags, on written "Artist" the other written "ARTIST". They both had different values so I manually removed the "ARTIST" tag from all affected files, re-indexed and the problem was solved.
